### PR TITLE
Cans 263 format validate case number

### PR DESCRIPTION
**Pull request by abelayt**
## Tests

- [x] I used TDD to develop the feature
- [x] I have included unit tests
- [ ] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I ran all my tests locally which were green.
- [x] My code adds no new issues to Code Climate
- [x] I ran all the linters for the project.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build (or bring donuts if I leave things broken), to check linters, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: app/javascript/Application/components/Client/ClientFormValidator.js [hunk 1]
```diff
@@ -4,7 +4,11 @@ export function validate(fieldName, value) {
     case 'last_name':
       return value.length && /^[A-z]+(([ ]|[-])[A-z]+)?$/i.test(value);
     case 'case_id':
-      return value.length && /^[a-zA-Z0-9]+$/i.test(value);
+      // return value.length > 18 && (/^\d{4}-\d{3}-\d{4}-\d{8}$/i.test(value) || /^\d{19}$/i.test(value));
```
Comment (jonbwalker) [inline app/javascript/Application/components/Client/ClientFormValidator.js]: Looks like we don't need line 7 anymore

File: app/javascript/Application/components/Client/ClientAddEditForm.js [hunk 2]
```diff
@@ -342,25 +342,38 @@ class ClientAddEditForm extends Component {
...
-              <TextField
-                required
-                focused
-                id="case_id"
-                label="Case Number"
-                defaultValue={childInfo.case_id}
-                error={!childInfoValidation['case_id']}
-                className={classes.textField}
+              <InputMask
+                mask="9999-999-9999-99999999"
```
Comment (jonbwalker) [inline app/javascript/Application/components/Client/ClientAddEditForm.js]: What do all these 9's do
What happens if you replace the 9's with something else?

Comment (abelayt) [reply app/javascript/Application/components/Client/ClientAddEditForm.js]: It is used to mask 0-9 strings for formatting characters.
'9' for '[0-9]'
'a': for '[A-Za-z]',

**Comment by jonbwalker** [comment]
Retest this please

**Pull request merged**